Good-Bids/goodbids
# [#364, #365] Email Variables and Email Triggers

# Summary

This PR sets up all the Email Classes and View Templates with the necessary variables and tokens to render the dynamic content when the emails get triggered.

## Issues

* #364 
* #365 
  * #236 
  * #247 
  * #237 
  * #238 
  * #248 
  * #239 
  * #249 

## Testing Instructions

1. Each Email has separate testing instructions depending on the type of email.
2. Locally, I was unable to get [MailPit](https://docs.wpvip.com/vip-local-development-environment/#h-optional-services) to work, so I would recommend installing [WP Mail Log](https://wordpress.org/plugins/wp-mail-log/) to catch emails sent by WP to verify the correct emails are sent at the correct triggers.
3. Triggers include:
    * Auction Start
    * Auction End
    * New Bid Placed
    * An Auction with an Unclaimed Reward has ended 7 or more days ago.

## Screenshots

**All Emails**
<img width="1070" alt="Screenshot 2024-02-20 at 8 58 17 AM" src="https://github.com/Good-Bids/goodbids/assets/122309362/e5a70687-cdd8-4e6a-8c96-be468de6557f">

**Auction Is Live**
<img width="633" alt="Screenshot 2024-02-20 at 4 04 26 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/4d0a071e-7dc0-45a7-ba10-0daf4679992e">

**Auction Is Live (Admin)**
<img width="616" alt="Screenshot 2024-02-20 at 5 10 24 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/3367887b-0d57-4cad-85e7-e224a262266c">

**Auction Closed**
<img width="616" alt="Screenshot 2024-02-20 at 4 46 27 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/a5002a63-5670-4025-9844-7919ce58b031">

**Auction Outbid**
<img width="625" alt="Screenshot 2024-02-20 at 5 13 34 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/f0fb81bb-1da2-4dee-9d31-1752fe267a80">

**Reward Reminder**
<img width="626" alt="Screenshot 2024-02-20 at 5 16 46 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/45033c88-2160-4099-ba14-6bf889e37263">

**Auction Summary (Admin)**
<img width="626" alt="Screenshot 2024-02-20 at 5 26 01 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/032e418c-307f-4412-830e-57ae6c5d4dba">

**Reward Confirmation**
<img width="625" alt="Screenshot 2024-02-20 at 5 28 33 PM" src="https://github.com/Good-Bids/goodbids/assets/122309362/136ddd7f-042a-4286-8c05-6106d77560f8">
